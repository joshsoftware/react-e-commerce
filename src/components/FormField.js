import React from 'react';
import { FormGroup } from 'reactstrap';
import FormInput from './FormInput';
import FormLabel from './FormLabel';
import FormFeedbackWrapper from './FormFeedbackWrapper';
import PropTypes from 'prop-types';

const FormField = ({ formfield }) => {
  const {
    field,
    labelText,
    type,
    name,
    placeholder,
    message,
    value,
    onChange,
    invalid,
    inputClassName,
    maxLength
  } = formfield;
  return (
    <FormGroup className={'align-items-center'}>
      <FormLabel field={field} labelText={labelText} />
      <FormInput
        type={type}
        name={name}
        placeholder={placeholder}
        value={value}
        onChange={onChange}
        invalid={invalid}
        className={inputClassName}
        maxLength={maxLength}
      />
      <FormFeedbackWrapper message={message} />
    </FormGroup>
  );
};

export default FormField;

FormField.propTypes = {
  formfield: PropTypes.object.isRequired
};
