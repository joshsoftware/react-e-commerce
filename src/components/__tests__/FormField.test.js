import { render, cleanup } from '@testing-library/react';
import React from 'react';
import FormField from '../FormField';

afterEach(cleanup);

describe('FormFild Component', () => {
  const formfield = {
    field: '',
    labelText: '',
    type: '',
    name: '',
    placeholder: '',
    message: ''
  };
  it('Must return FormField Component', () => {
    const { asFragment } = render(<FormField formfield={formfield} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
