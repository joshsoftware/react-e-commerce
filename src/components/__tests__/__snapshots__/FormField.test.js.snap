// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormFild Component Must return FormField Component 1`] = `
<DocumentFragment>
  <div
    class="form-group"
  >
    <label
      class=""
      for=""
    />
    <input
      class="form-control"
      name=""
      placeholder=""
      type=""
      value=""
    />
    <div
      class="invalid-feedback"
    />
  </div>
</DocumentFragment>
`;
